pa/3.md

```diff
@@ -25,11 +25,29 @@ The assignment has two parts. In the first, you will implement a copying garbage
 
 Make sure your VM from the previous assignment is in working order. If you need help, come talk to us in office or lab hours.
 
-## Part I: Garbage Collection
+## Part I: Reference Counting
+
+Extend your VM from PA2 to perform reference counting. Refer to Appel 13.2 and the lecture notes to review reference counting.
+
+For **Part I**, your VM extended with reference counting will have the same interface as your VM from PA2 except that it will keep track of references to objects on the heap during execution. Specifically, it should:
+
+1. Accept the name of a GrumpyVM bytecode `<filename.o>` as its first command-line argument;
+2. Execute the file, using reference counting for any data on the heap;
+3. Print the program's result to `stdout` using the textual encoding described in PA2 (`Vi32(-3)` for the integer `-3`, etc.). 
+
+More specific to reference counting, you will also need to meet the following guidelines:
+
+4. Write at least 2 unit tests specific to reference counting:
+    - One should verify that your reference counting implementation removes data with 0 references
+    - One should verify that the data you clear from the heap is used by Alloc if a spot exists with enough spaces
+5. When clearing spots from the heap, do not change the size of the heap, reference counting does not de-fragment memory
+6. If your VM cannot complete allocation, even with the space freed by the reference counter, then your VM should exit with a nonzero exit code
```

Comment:
We should say something here about the fact that they don't need to collect cycles.

Reply:
Does the most recent addition work, or should it be a little more clear?